uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestStatusMatrixTest.php

```diff
@@ -266,9 +266,9 @@ public function getManagerExclusivePossibleStatusTransitionsDataProvider() {
     $possibleStaffTransitions = $this->allPossibleStatusTransitionForStaffDataProvider();
     $possibleManagerTransitions = $this->allPossibleStatusTransitionForLeaveApprover();
 
-    $results = array_diff(array_map('serialize', $possibleManagerTransitions), array_map('serialize', $possibleStaffTransitions));
-    $managerExclusiveStatusTransition = array_map('unserialize', $results);
-
-    return $managerExclusiveStatusTransition;
+    $comparator = function ($a, $b) {
+      return strcmp(sprintf('%d-%d', $a[0], $a[1]), sprintf('%d-%d', $b[0], $b[1]));
+    };
+    return array_udiff($possibleManagerTransitions,$possibleStaffTransitions, $comparator);
```

Comment:
You're probably going to hate me, but I did a check of the performance for a small array.

```
100000 runs:

serialization: 1.0214629173279
udiff: 3.2251260280609
```

Maybe for large, multidimensional arrays `udiff` could be faster, but if the arrays are all just arrays of 2 integers it seems like serialization has the edge.